substreams/tests/parser_test.rs

```diff
@@ -0,0 +1,50 @@
+use rstest::rstest;
+use substreams::parser::{evaluate_expression};
+ 
+static TEST_KEYS: &[&str] = &["test", "test1", "test2", "test3", "test4", "test5", "test 6"];
+
+#[rstest]
+#[case(TEST_KEYS, "test", true)]
+#[case(TEST_KEYS, "'test'", true)]
+#[case(TEST_KEYS, "'test 6' || test7", true)]
+#[case(TEST_KEYS, "'test_6' && test3", false)]
+#[case(TEST_KEYS, "\"test 6\" || test7", true)]
+#[case(TEST_KEYS, "\"test 6\" && test3", true)]
+
+#[case(TEST_KEYS, "test1 || test", true)]
+#[case(TEST_KEYS, "test1 || test6", true)]
+#[case(TEST_KEYS, "test6 || test7", false)]
+
+#[case(TEST_KEYS, "test1 || test || test2", true)]
+#[case(TEST_KEYS, "test1 || test6 || test7", true)]
+#[case(TEST_KEYS, "test6 || test7 || test8", false)]
+
+#[case(TEST_KEYS, "test1 && test", true)]
+#[case(TEST_KEYS, "test1 && test6", false)]
+#[case(TEST_KEYS, "test6 && test7", false)]
+
+#[case(TEST_KEYS, "test1 && test && test2", true)]
+#[case(TEST_KEYS, "test1 && test2 && test7", false)]
+#[case(TEST_KEYS, "test6 && test7 && test8", false)]
+
+#[case(TEST_KEYS, "test1 test", true)]
+#[case(TEST_KEYS, "test1 test6", false)]
+#[case(TEST_KEYS, "test6 test7", false)]
+
+#[case(TEST_KEYS, "(test1)", true)]
+#[case(TEST_KEYS, "(test1 test6)", false)]
+
+#[case(TEST_KEYS, "test1     test2 ", true)]
+#[case(TEST_KEYS, "test1    && test2 ", true)]
+#[case(TEST_KEYS, "test1    &&     test6", false)]
```

Comment:
How do you validate that this is parsed correctly? It parses ok, but the resulting expression should be checked.

In the Go version, I created a string representation of the tree structure, so it can be verified too.